Ryujinx.Graphics.Vulkan/FenceHelper.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Silk.NET.Vulkan;
+using System;
+
+namespace Ryujinx.Graphics.Vulkan
+{
+    static class FenceHelper
+    {
+        private const ulong DefaultTimeout = 100000000; // 100ms
+
+        public static bool AnySignaled(Vk api, Device device, ReadOnlySpan<Fence> fences, ulong timeout = 0)
+        {
+            return api.WaitForFences(device, (uint)fences.Length, fences, false, timeout) == Result.Success;
+        }
+
+        public static bool AllSignaled(Vk api, Device device, ReadOnlySpan<Fence> fences, ulong timeout = 0)
+        {
+            return api.WaitForFences(device, (uint)fences.Length, fences, true, timeout) == Result.Success;
+        }
+
+        public static void WaitAllIndefinitely(Vk api, Device device, ReadOnlySpan<Fence> fences)
+        {
+            Result result;
+            while ((result = api.WaitForFences(device, (uint)fences.Length, fences, true, DefaultTimeout)) == Result.Timeout)
+            {
+                // Keep waiting while the fence is not signaled.
+            }
```

Comment:
Not sure if I like the while loop or the do..while loop better, up to you :shrug: 
```suggestion
            do
            {
                // Keep waiting while the fence is not signaled.
                result = api.WaitForFences(device, (uint)fences.Length, fences, true, DefaultTimeout);
            }
            while (result == Result.Timeout);
```

Reply:
I don't like do-while loops, so I avoid using them as much as possible. One can place the while after the end bracket like `} while` but then it feels inconsistent since we don't do `while {`. One can put it on the next line, but then it looks weird and does not match what other projects do. It's just a silly and personal annoyance, I guess.

Reply:
Good enough for me, happy to leave it how it is.